### PR TITLE
[rc] Replace `HttpObserve`

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Angular removed `HttpObserve` from their `@angular/common/http` API.
See: https://github.com/angular/angular/issues/18146#issuecomment-497339251

This branch also adds support for Angular 9 and Builder `4-rc`.

### CODE REVIEW

File: src/app/public/modules/auth-http-client/auth-options.ts [hunk 1]
```diff
@@ -20,7 +16,7 @@ import {
 export function skyAuthHttpOptions(options?: {
   body?: any,
   headers?: HttpHeaders,
-  observe?: HttpObserve,
+  observe?: 'body'|'events'|'response',
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/auth-http-client/auth-options.ts:19]: See: https://github.com/angular/angular/issues/18146#issuecomment-497339251

File: package.json [hunk 2]
```diff
@@ -22,53 +22,52 @@
...
-    "@angular/forms": "7.2.15",
-    "@angular/http": "7.2.15",
-    "@angular/platform-browser": "7.2.15",
-    "@angular/platform-browser-dynamic": "7.2.15",
-    "@angular/router": "7.2.15",
-    "@blackbaud/auth-client": "2.19.0",
-    "@pact-foundation/pact": "9.5.0",
-    "@pact-foundation/pact-web": "9.5.0",
-    "@skyux-sdk/builder": "3.11.0",
+    "@angular/animations": ">9.0.0-rc < 10.0.0",
```
Comment (Blackbaud-BobbyEarl) [inline package.json]: What's the reasoning for switching from explicit versions in `devDependencies`?

File: package.json [hunk 3]
```diff
@@ -22,53 +22,52 @@
   },
   "homepage": "https://github.com/blackbaud/skyux-http#readme",
   "peerDependencies": {
-    "@angular/common": ">=4.3.6",
-    "@angular/core": ">=4.3.6",
-    "@blackbaud/auth-client": "^2.17.1",
+    "@angular/common": ">9.0.0-rc < 10.0.0",
```
Comment (Blackbaud-BobbyEarl) [inline package.json]: Do we know for sure `10.x` wouldn't supported?  I'm not necessarily opposed, but being explicit about not supporting future versions is a departure in our current procedures.

Comment (Blackbaud-SteveBrush) [reply package.json]: Good question. Is it better to support all future versions or only a specific known subset? I've seen both setups in various packages. If SKY UX's stance is to support all future versions, I can change it.

File: package.json (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply package.json]: Good point. I'll revert it back. I wanted the latest RC version, but this goes against our existing standards.

File: package.json (hunk 3 above)
Comment (Blackbaud-BobbyEarl) [reply package.json]: My vote would be to support it until we know it doesn't work.  Given our frequency is slower than that of Angular, that has bought us a few extra major versions in the past.

**Review by Blackbaud-BobbyEarl — APPROVED** [review]

**Pull request merged**